scipy/stats/_stats_py.py

```diff
@@ -717,8 +713,12 @@ def tmin(a, lowerlimit=None, axis=0, inclusive=True, nan_policy='propagate'):
     14
 
     """
+    dtype = a.dtype
     a = _put_nan_to_limits(a, (lowerlimit, None), (inclusive, None))
-    return np.nanmin(a, axis=axis)
+    res = np.nanmin(a, axis=axis)
+    if not np.any(np.isnan(res)):
+        return res.astype(dtype)
```

Comment:
To avoid a copy if the dtype is already the same as the input:
```suggestion
        # needed if input is of integer dtype
        return res.astype(dtype, copy=False)
```